tasks/javascript.js

```diff
@@ -113,6 +115,17 @@ gulp.task('javascript:lint', function () {
 
 // karma unit tests in local library only
 gulp.task('javascript:unit', function () {
+  var reporters = cliArgs.karmareporter;
+  if (reporters.constructor === Array) {
+    reporters = Array;
```

Comment:
I do not follow what is happening here. Why assign a class `Array` when all other permutations are values.
If this line is as intended then you need some explanation as to what is going on.